tests/ocp/sriov/README.md

```diff
@@ -0,0 +1,557 @@
+    
+    // Namespaces
+    TestNamespaceName = "sriov-tests"
+    
+    // Timeouts
+    WaitTimeout = 3 * time.Minute
+    DefaultTimeout = 300 * time.Second
+)
+
+var (
+    // Labels list for test selection
+    Labels = []string{LabelSuite, LabelFeatureName}
+    
+    // Reporter configuration
+    ReporterCRDsToDump = []k8sreporter.CRData{...}
+    ReporterNamespacesToDump = map[string]string{...}
+)
+```
+
+#### Best Practices
+
+1. **Constants**: Use for values that don't change (labels, namespaces, default values)
+2. **Variables**: Use for values that may be configured or computed
+3. **Timeouts**: Define reasonable defaults; make them configurable when needed
+4. **Labels**: Aggregate in a `Labels` variable for suite-level filtering
+
+### Environment Validation Package (`sriovenv` or similar)
+
+Purpose: Validate cluster state and test prerequisites.
+
+#### Responsibilities
+
+- Verify operator deployment status
+- Validate available hardware resources
+- Check cluster configuration
+- Discover and validate test interfaces
+
+Example:
+```go
+func ValidateSriovInterfaces(nodeList []*nodes.Builder, minCount int) error {
+    // Validation logic
+}
+
+func IsSriovDeployed(apiClient *client.Client, config *netconfig.Config) error {
+    // Deployment check
+}
+```
+
+### Code Organization and Reusability
+
+#### Reusable Functions Location
+
+**Important**: Reusable functions that can be used across multiple test suites should be placed in the upper-level `internal/` folder (e.g., `tests/internal/`), not in the test suite-specific `internal/` folder.
+
+1. **Suite-specific helpers**: Place in `tests/ocp/sriov/internal/`
+   - Functions specific to SR-IOV test suite only
+   - Test parameters and constants specific to this suite
+
+2. **Reusable/common helpers**: Place in `tests/internal/`
+   - Functions that can be used by multiple test suites
+   - Common utilities and helpers
+   - Shared validation logic
+
+Example structure:
+```
+tests/
+├── internal/                    # Reusable across all test suites
+│   ├── cluster/                # Cluster-level utilities
+│   ├── params/                 # Common parameters
+│   └── reporter/               # Common reporting utilities
+└── ocp/
+    └── sriov/
+        └── internal/            # SR-IOV suite-specific only
+            ├── tsparams/       # SR-IOV test parameters
+            └── sriovenv/        # SR-IOV environment validation
+```
+
+## Test Implementation Rules
+
+### Error Handling
+
+1. **Always check errors**: Don't ignore errors; handle them appropriately
+2. **Use Gomega matchers**: Prefer `Expect(err).ToNot(HaveOccurred())` for assertions
+3. **Descriptive error messages**: Include context in error messages
+4. **Cleanup on failure**: Use `AfterEach` or `DeferCleanup` for resource cleanup
+
+### Resource Management
+
+1. **Create resources in BeforeAll/BeforeEach**: Set up test prerequisites
+2. **Cleanup in AfterAll/AfterEach**: Always clean up created resources
```

Comment:
We also need to recover any existing CRs that we deleted if applicable, probably at BeforeAll/AfterAll or BeforeSuite/AfterSuite level. e.g., capture existing CRs at BeforeXX and recover them at AfterXX.